src/vs/workbench/api/common/extHostSearch.ts

```diff
@@ -37,8 +38,27 @@ export class ExtHostSearch implements ExtHostSearchShape {
 	constructor(
 		@IExtHostRpcService private extHostRpc: IExtHostRpcService,
 		@IURITransformerService protected _uriTransformer: IURITransformerService,
-		@ILogService protected _logService: ILogService
-	) { }
+		@ILogService protected _logService: ILogService,
+		commands: ExtHostCommands,
+	) {
+		commands.registerArgumentProcessor({
+			processArgument: arg => {
+				if (arg && Array.isArray(arg)) {
```

Comment:
There is a lot of indentation here.  Could we switch the logic and return early. e.g.
`if (!arg || !Array.isArray(arg)) {
   return arg;
}'

This removes the needs for the else and makes the code a little more readbale.

Same could be done to the === 13 line.